### ISSUE TITLE
About the dosage of loop diuretics in inputevents_cv

### ISSUE THREAD

**Issue by ditto504** [MIT-LCP/mimic-code#322]
Hi, I am totally confused about the amount and rate in inputevents_cv when I  tried to find the dosage of Lasix, a loop diuretic drug. I have read all the documents about input/output of CareVue and I know the amount and the rate are asynchronous in inputevents_cv.

Following is some rows of 'itemid=30123(Lasix) and icustay=289357 order by charttime':

|line |charttime|	itemid|	amount|	rate|	|	storetime|	originalamout||	originalroute|
| ------------- |:-------------:| -----:|-----:|-----:|-----:|-----:|-----:|-----:|-----:|
some lines...
|1| 9/30 19:00|	30123|	0|	mg|	|	9/30 22:11|	100|	mg|	Intravenous Push|	|
|2| 9/30 20:00|	30123|	|	30|	mghr|	9/30 22:11|	||	IV Drip|	|
|3 |9/30 20:00|	30123|	30|	mg|	|	9/30 22:11|	100|	mg|	Intravenous Push|	|
|4| 9/30 21:00|	30123|	|	30|	mghr|	9/30 22:11|	||	IV Drip|	|
|5| 9/30 21:00|	30123|	120|	mg|	|	9/30 22:11|	400|	mg|	Intravenous Push|	7.5|
|6| 9/30 22:00|	30123|	120|	mg|	|	9/30 22:11|	400|	mg|	Intravenous Push|	7.5|
|7| 9/30 22:00|	30123|	|	30|	mghr|	9/30 22:11|	||	IV Drip|	|
|8| 9/30 23:00|	30123|	|	30|	mghr|	10/1 1:55|	||	IV Drip|	|
|9| 9/30 23:00|	30123|	120|	mg|	|	10/1 1:55|	400|	mg|	Intravenous Push|	7.5|
some lines...
||10/1 11:00|	30123|	30|	mg|	|	10/1 11:05|	400|	mg|	Intravenous Push|	7.5|
||10/1 11:00|	30123|	|	30|	mghr|	10/1 11:05|	||	IV Drip|	|
||10/1 13:00|	30123|	|	30|	mghr|	10/1 13:28|	||	IV Drip|	|
||10/1 14:00|	30123|	|	30|	mghr|	10/1 14:01|	||	IV Drip|	|
||10/1 14:15|	30123|	|	30|	mghr|	10/1 15:14|	||	IV Drip|	|

If I understand correct, in line #1, the amount of 0 is since the previous amount order by charttime. So I guess the in line #3, the amount of 30 is calculated from a previous 30mghr rate start of 20:00 (not show in above). But how to explain the 120mg in line #5 and line #6? The rate start from 20:00 and 21:00 are both 30 mg/hr, it is not possible to get two 120mg in the following 2 hour.

Another question is, I wish to compare different administration route (bolus/intravenous push vs. continuous infusion/IV drip), how can I tell which line is a bolus rather than just an amount record?

Thanks a lot.

**Comment by alistairewj** [comment]
Looks like the bag was switched from a 100mg bag to a 400mg bag, and so the amount of drug delivered increased accordingly. There is a bit of inconsistency between rate/amount because of the way the database archived the data so you have to work around this sort of data.

They do record administration route (`originalroute`) but it's not very consistent in its terminology. For example in this case it lists it as an IV Drip for the rate and as an Intravenous Push for the amount. In the MetaVision database system you can identify bolus administrations as those with a duration of 1 minute. In the CareVue database system I'm not sure of a good systematic way of identifying boluses.

**Comment by ditto504** [comment]
Thanks a lot @alistairewj .

But I still don't understand. The `rate` unit is mg/hr, when changing from a 100mg bag into a 400mg bag, if the rate is under 30 mg/hr, there will be no difference of how many mg in an hour. Unless the `rate` is not mg/hr?

It is really strange that I can only see IV drip under mghr of lasix in CareVue. Usually in practice, we use bolus quite often.

Besides, I have reviewed the lasix/furosemide duration in `inputevents_cv` using sql codes from [dopamin-durations.sql](https://github.com/MIT-LCP/mimic-code/blob/master/concepts/durations/dopamine-durations.sql). I wanted to find those with short duration (i.e. 1min or 5min), but It dosen't work.

If I cannot extract bolus/continuous infusion from CareVue, we have to use MetaVision only. What a pity:(

**Comment by alistairewj** [comment]
The rate is inaccurate in this case. Not sure about the general documentation practice for a lasix bolus - perhaps they document it in an unusual way.

The durations query that you are using makes the assumption of a continuous infusion, and so it combines sequential recordings of the same drug into one duration recording. This makes sense for vasopressors which are only given as a continuous infusion but may not make sense for lasix boluses.

Using CareVue is a bit more challenging - up to you whether you use data from both sources or not.

**Comment by ditto504** [comment]
Thank you and I have decided to use MetaVision alone. I usedR to calculate each patient's amount of using diuretics including bolus and continuous infusion during specific duration ( e.g. every 12 hours or 24hours) of each icustay.

Btw, @alistairewj . I don't know whether it is okay to ask the question about `eICU` here. Is that possible identifying a bolus or continuous infusion drug in `eICU`?

Is the drug recorded in `infusiondrug` as continuous infusion? But I found some of the drug in the table `infusiondrug` have `infusionrate` and others are not, can I say those `infusionrate is not null` as continuous infusion? And how can I know how long of this infusion really administrated? Is the same iv drug recorded both in table `medication` and `infusiondrug`? The documentation is not quite detail in eICU.
